cmd/prerun.go

```diff
@@ -0,0 +1,94 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/Masterminds/semver"
+	"github.com/astronomer/astro-cli/deployment"
+	"github.com/astronomer/astro-cli/houston"
+	"github.com/astronomer/astro-cli/messages"
+	"github.com/astronomer/astro-cli/version"
+	"github.com/spf13/cobra"
+)
+
+// CheckDeploymentVersion for validation of the CLI vs Deployment Version
+func CheckDeploymentVersion(client *houston.Client, cmd *cobra.Command, out io.Writer) {
+	appCfg := deployment.GetAppConfig()
+	cv := version.CurrVersion
+
+	// Skip check if AppConfig is nil or is cv is empty
+	if appCfg != nil && cv != "" {
+		dv := appCfg.Version
+		validateVersions(cv, dv, out)
```

Comment:
```suggestion
		validateVersions(version.CurrVersion, appCfg.Version, out)
```
no need this redundant lines, you are just waste of memory here by copying values.